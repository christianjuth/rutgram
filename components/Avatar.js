import React from 'react';
import PropTypes from 'prop-types';
import { View, StyleSheet, TouchableOpacity } from 'react-native';
import { Avatar as PaperAvatar } from 'react-native-paper';
import { LinearGradient } from 'expo-linear-gradient';

function Avatar(props) {
  let createCircleStyle = (size) => {
    return {
      height: size,
      width: size,
      borderRadius: size/2
    };
  };

  let size = props.size || 50,
      spacerSize = size * 1.08,
      gradientSize = size * 1.15;

  return(
    <TouchableOpacity
      onPress={props.onPress}
      activeOpacity={1}
    >
      <LinearGradient
        colors={['#12C2E9', '#C471ED', '#F64F59']}
        style={[styles.gradient, createCircleStyle(gradientSize)]}
      >
        <View style={[styles.spacer, createCircleStyle(spacerSize)]}>
          <PaperAvatar.Image {...props} size={size}/>
        </View>
      </LinearGradient>
    </TouchableOpacity>
  );
}

Avatar.propTypes = {
  size: PropTypes.number,
  onPress: PropTypes.func
};

const styles = StyleSheet.create({
  gradient: {
    alignItems: 'center',
    justifyContent: 'center',
  },

  spacer: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#fff'
  }
});

export default Avatar;
