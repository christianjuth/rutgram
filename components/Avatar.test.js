import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Avatar from './Avatar';

Enzyme.configure({ adapter: new Adapter() });

it('renders correctly', () => {
  const tree = renderer
    .create(<Avatar size={40}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('double tap', () => {
  const mockFunc = jest.fn();
  const component = shallow(<Avatar onPress={mockFunc}/>);
  component.simulate('press');
  expect(mockFunc).toHaveBeenCalled();
});
