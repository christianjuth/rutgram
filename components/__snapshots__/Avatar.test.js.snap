// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4279419625,
          4291064301,
          4294332249,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Object {
          "borderRadius": 23,
          "height": 46,
          "width": 46,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "justifyContent": "center",
          },
          Object {
            "borderRadius": 21.6,
            "height": 43.2,
            "width": 43.2,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#6200ee",
              "borderRadius": 20,
              "height": 40,
              "width": 40,
            },
            undefined,
          ]
        }
      >
        <Image
          style={
            Object {
              "borderRadius": 20,
              "height": 40,
              "width": 40,
            }
          }
        />
      </View>
    </View>
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;
